src/handlers/push_message.rs

```diff
@@ -358,10 +362,29 @@ pub async fn handler_internal(
         "fetched provider"
     );
 
-    provider
-        .send_notification(client.token, body.payload)
-        .await
-        .map_err(|e| (e, analytics.clone()))?;
+    match provider.send_notification(client.token, body.payload).await {
+        Ok(_) => Ok(()),
+        Err(error) => match error {
+            Error::BadDeviceToken => {
+                state.client_store.delete_client(&tenant_id, &id);
```

Comment:
is this a soft delete?

Reply:
No its a hard delete as the data we have stored is now useless, we still have data about the client in datalake